Model/Product/CategoryProcessor.php

```diff
@@ -1,108 +1,93 @@
-     */
-    protected $categoryColFactory;
-    
     /**
      * Categories id to object cache.
      *
      * @var array
      */
     protected $categoriesCache = [];
-    
+
+    /**
+     * @var \Magento\Catalog\Model\ResourceModel\Category\CollectionFactory
+     */
+    protected $categoryColFactory;
+
     /**
-     * Instance of catalog category factory.
-     *
      * @var \Magento\Catalog\Model\CategoryFactory
      */
     protected $categoryFactory;
-    
+
+    /**
+     * @var CategoryUrlPathGenerator
+     */
+
+    protected $categoryUrlPathGenerator;
+
     /**
      * @var CategoryUrlRewriteGenerator
      */
     protected $categoryUrlRewriteGenerator;
-    
+
     /**
      * @var UrlPersistInterface
      */
     protected $urlPersist;
-    
+
     /**
+     * CategoryProcessor constructor.
      * @param \Magento\Catalog\Model\ResourceModel\Category\CollectionFactory $categoryColFactory
-     * @param \Magento\Catalog\Model\CategoryFactory                          $categoryFactory
+     * @param \Magento\Catalog\Model\CategoryFactory $categoryFactory
+     * @param CategoryUrlPathGenerator $categoryUrlPathGenerator
+     * @param CategoryUrlRewriteGenerator $categoryUrlRewriteGenerator
+     * @param UrlPersistInterface $urlPersist
+     * @throws \Magento\Framework\Exception\LocalizedException
+     * @throws \Magento\UrlRewrite\Model\Exception\UrlAlreadyExistsException
      */
     public function __construct(
         \Magento\Catalog\Model\ResourceModel\Category\CollectionFactory $categoryColFactory,
         \Magento\Catalog\Model\CategoryFactory $categoryFactory,
         CategoryUrlPathGenerator $categoryUrlPathGenerator,
-        ChildrenCategoriesProvider $childrenCategoriesProvider,
-        StoreViewService $storeViewService,
         CategoryUrlRewriteGenerator $categoryUrlRewriteGenerator,
         UrlPersistInterface $urlPersist
     ) {
-        $this->categoryColFactory          = $categoryColFactory;
-        $this->categoryFactory             = $categoryFactory;
-        $this->categoryUrlPathGenerator    = $categoryUrlPathGenerator;
-        $this->childrenCategoriesProvider  = $childrenCategoriesProvider;
-        $this->storeViewService            = $storeViewService;
+        $this->categoryColFactory = $categoryColFactory;
+        $this->categoryFactory = $categoryFactory;
         $this->categoryUrlRewriteGenerator = $categoryUrlRewriteGenerator;
-        $this->urlPersist                  = $urlPersist;
+        $this->categoryUrlPathGenerator = $categoryUrlPathGenerator;
+        $this->urlPersist = $urlPersist;
+
         $this->initCategories();
     }
 
     /**
      * @return $this
-     * @throws \Exception
+     * @throws \Magento\Framework\Exception\LocalizedException
      * @throws \Magento\UrlRewrite\Model\Exception\UrlAlreadyExistsException
      */
     protected function initCategories()
     {
+        set_time_limit(0);
```

Comment:
set_time_limit is bad practice, not to mention unenforceable. If this task is running from CLI, like any long running task should be, then there is no time limit, and if the task is running in a PHP-FPM process it will be killed by the process manager regardless of set_time_limit after 300 seconds (on turnkeys, possibly shorter on other hosts)